hibernate-core/src/main/java/org/hibernate/dialect/identity/H2IdentityColumnSupport.java

```diff
@@ -23,11 +23,22 @@ public String getIdentityColumnString(int type) {
 
 	@Override
 	public String getIdentitySelectString(String table, String column, int type) {
+		//not supported since H2 1.4.200
 		return "call identity()";
 	}
 
 	@Override
 	public String getIdentityInsertString() {
-		return "null";
+		return "default";
+	}
+
+	@Override
+	public boolean supportsInsertSelectIdentity() {
+		return true;
+	}
+
+	@Override
+	public String appendIdentitySelectToInsert(String insertString) {
+		return "select id from final table (" + insertString + ")";
```

Comment:
I explored something like this. This does not work if the column is not named `id`

```
19:03:24.089 [Time-limited test] ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper - Column "ID" not found; SQL statement:
select id from final table (insert into Event (id2, active, createdOn) values (default, ?, ?)) [42122-202]
```

Reply:
OK thanks.